src/libcork/pthreads/thread.c

```diff
@@ -116,17 +124,30 @@ cork_thread_get_id(struct cork_thread *self)
     return self->id;
 }
 
+#define PTHREADS_MAX_THREAD_NAME_LENGTH  16
+
 static void *
 cork_thread_pthread_run(void *vself)
 {
     int  rc;
     struct cork_thread  *self = vself;
     struct cork_thread_descriptor  *desc = cork_thread_descriptor_get();
+#if defined(__APPLE__) && defined(__MACH__)
+    char  thread_name[PTHREADS_MAX_THREAD_NAME_LENGTH];
+#endif
 
     desc->current_thread = self;
     desc->id = self->id;
     rc = self->run(self->user_data);
 
+#if defined(__APPLE__) && defined(__MACH__)
```

Comment:
It's confusing to see the same (platform dependent) code in two separate functions, one a callback used in the other.